src/components/Layout/Carousel/Carousel.vue

```diff
@@ -0,0 +1,128 @@
+<template>
+  <div class="swiperContainer">
+    <carousel-progress-bar
+      v-if="progressBar && topProgressBar && slides > 0"
+      :pages="slides"
+      :current-index="currentIndex"
+      :duration="slideDuration"
+      :paused="isPaused"
+      class="px-2 px-sm-4 progress-bar"
+      hoverable
+      @on-animation-end="onAnimationEnd"
+      @on-progress-clicked="onProgressClicked"
```

Comment:
For now it's not a bother, but I'm sure we'll be able to factorize the top and bottom progress bars